src/main/java/uk/gov/hmcts/darts/task/runner/impl/UnstructuredDataExpiryDeletionAutomatedTask.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Override
+    public AutomatedTaskName getAutomatedTaskName() {
+        return AutomatedTaskName.UNSTRUCTURED_DATA_EXPIRY_DELETION_TASK_NAME;
+    }
+
+    @Override
+    public void runTask() {
+        final UserAccountEntity userAccount = userIdentity.getUserAccount();
+        OffsetDateTime maxRetentionDate = currentTimeHelper.currentOffsetDateTime();
+        Limit batchSize = Limit.of(getAutomatedTaskBatchSize());
+
+        deleteTranscriptionDocumentEntity(userAccount, maxRetentionDate, batchSize);
+        deleteMediaEntity(userAccount, maxRetentionDate, batchSize);
+        deleteAnnotationDocumentEntity(userAccount, maxRetentionDate, batchSize);
+        deleteCaseDocumentEntity(userAccount, maxRetentionDate, batchSize);
+    }
+
+    void deleteTranscriptionDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            transcriptionDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredTranscriptionDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getTranscriptionDocumentEntity
+        );
+    }
+
+
+    void deleteMediaEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            mediaRepository,
+            externalObjectDirectoryRepository.findExpiredMediaEntries(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getMedia
+        );
+    }
+
+    void deleteAnnotationDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            annotationDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredAnnotationDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getAnnotationDocumentEntity
+        );
+    }
+
+    void deleteCaseDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            caseDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredCaseDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getCaseDocument
+        );
+    }
+
+
+    <T extends SoftDelete> void deleteExternalObjectDirectoryEntity(
+        UserAccountEntity userAccount,
+        JpaRepository<T, Integer> repository,
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities,
+        Function<ExternalObjectDirectoryEntity, T> entityMapper) {
+
+
+        List<ExternalObjectDirectoryEntity> deletedEntities = externalObjectDirectoryEntities
+            .stream()
+            .filter(this::deleteFromExternalDataStore)
+            .toList();
+
+        repository.saveAll(
```

Comment:
Optional but we could introduce a soft repository interface that soft deletes all soft entities and saves them

Reply:
Agreed updated :)